### PR TITLE
Update README, create proper CHANGELOG and LICENSE files

### LINKED ISSUE #37: README is highly out of date
It should be updated before release, as well as a change log added.

### PR DESCRIPTION
**Pull request by le717**
Fixes #37.

@MiguelCastillo I hate to bug you again, but would you mind too terribly to review and wordsmith this for me as well? :)

### CODE REVIEW

File: styles/whitespace-colors-css.tmpl [hunk 1]
```diff
@@ -7,8 +7,8 @@
 .CodeMirror .cm-dk-whitespace-tab:before { background-color: <%= light.whitespace %>; }
 .CodeMirror .cm-dk-whitespace-leading-space:before,
 .CodeMirror .cm-dk-whitespace-leading-tab:before { background-color: <%= light.leading %>; }
-.CodeMirror .cm-dk-whitespace-empty-line-tab:before,
-.CodeMirror .cm-dk-whitespace-empty-line-space:before { background-color: <%= light.empty %>; }
+.CodeMirror .cm-dk-whitespace-empty-line-space:before,
+.CodeMirror .cm-dk-whitespace-empty-line-tab:before { background-color: <%= light.empty %>; }
```
Comment (le717) [inline styles/whitespace-colors-css.tmpl]: These were simply in the wrong order. The rules are written in space-tab order, but this and the dark version was in tab-space order.

**Comment by MiguelCastillo** [comment]
of course! :)

File: README.md [hunk 2]
```diff
@@ -1,32 +1,25 @@
...
+## Usage
 
-The styles are defined in `main.less` which is compiled to CSS and loaded into Brackets when loading the extension.
+Activate using `View > Show Whitespace` or by using the shortcut `Ctrl-Shift-W` (`Command-Shift-W` on Mac OS X). The extension state is remembered across Brackets launches. Whitespace in all editor windows are visualized, including inline editors.
 
-Whitespace in inline editors is visualized as well.
+This extension uses [CodeMirror](http://codemirror.net/) overlays to construct and render whitespace for both spaces and tabs. Each individual blank character are wrapped in a `<span></span>` with an appropriate denoting class. The following are all the classes this extension uses and their purpose. Do note that tabs follow this same structure, with the word `space` replaced with `tab`.
```
Comment (le717) [inline README.md]: Just realized I might want a header here to indicate this is technical details. How about "Technical Details" or "How It Works"?

**Comment by le717** [comment]
Miguel, do you want to quickly look at this while you're online? :)

File: README.md [hunk 3]
```diff
@@ -1,32 +1,31 @@
-# Show Whitespace in Brackets
+# Show Whitespace
 
-[Brackets](http://brackets.io/) extension to visualize whitespace (spaces and tabs) like [Sublime Text](http://www.sublimetext.com/) does. Adds an entry called "Show Whitespace" to the View menu, the shortcut is Command-Shift-W (or Ctrl-Shift-W on Windows). The state is remembered next time you start Brackets.
+> A [Brackets](http://brackets.io/) extension to visualize whitespace (both spaces and tabs) in the same style as [Sublime Text](http://www.sublimetext.com/).
```
Comment (MiguelCastillo) [inline README.md]: adding sublime text does not really seem to add much value to... but maybe that's my biased perception :)

File: README.md [hunk 4]
```diff
@@ -1,32 +1,31 @@
...
+## Usage
 
-The styles are defined in `main.less` which is compiled to CSS and loaded into Brackets when loading the extension.
+Activate using `View > Show Whitespace` or by using the shortcut `Ctrl-Shift-W` (`Command-Shift-W` on Mac OS X). The extension state is remembered across Brackets launches. Whitespace in all editor windows are visualized, including inline editors.
 
-Whitespace in inline editors is visualized as well.
+## Technical Details
 
+This extension uses [CodeMirror](http://codemirror.net/) overlays to construct and render whitespace for both spaces and tabs. Each individual blank character are wrapped in a `<span></span>` with an appropriate denoting class. The following are all the classes this extension uses and their purpose. Do note that tabs follow this same structure, with the word `space` replaced with `tab`.
```
Comment (MiguelCastillo) [inline README.md]: small correction:  "Each individual character *is* wrapped in a `<span></span>`"...

File: package.json [hunk 5]
```diff
@@ -1,15 +1,13 @@
 {
     "name": "dkehrig.show-whitespace",
     "title": "Show Whitespace",
-    "description": "Visualizes whitespace similar to how Sublime Text does it",
+    "description": "Visualize whitespace (both spaces and tabs) in the same style as Sublime Text. Now with configurable whitespace colors!",
```
Comment (le717) [inline package.json]: I don't know about others, but I often use the description to highlight very notable changes in the new and/or previous few versions.

File: README.md [hunk 6]
```diff
@@ -1,32 +1,31 @@
...
+* `.cm-dk-whitespace-leading-space`: Whitespace that makes up any indentation
+* `.cm-dk-whitespace-empty-line-space`: Any lines that consist solely of whitespace
+* `.cm-dk-whitespace-trailing-space`: Whitespace at the end of a line after any characters
 
+The primary extension styling is defined in `styles/main.less`, which is compiled into CSS, while whitespace colors are defined in Brackets preferences and rendered into `styles/whitespace-colors-css.tmpl`. Both files are then loaded into Brackets on startup.
 
-## Changelog
-
-### 2012-11-14
+## Install
```
Comment (MiguelCastillo) [inline README.md]: What about just including a link to brackets' wiki for installation instructions?  https://github.com/adobe/brackets/wiki/Brackets-Extensions#installing-and-removing-extensions

File: README.md (hunk 3 above)
Comment (le717) [inline README.md]: The original readme mentioned Sublime Text, so I kept it in.

File: README.md [hunk 7]
```diff
@@ -1,32 +1,31 @@
...
-Changed the name to "Show Whitespace" as all whitespace is visualized now.
+Method 1: Open the Brackets Extension Manager and search for "show whitespace"
 
-### 2012-06-29
+Method 2: Download the [latest revision](https://github.com/DennisKehrig/brackets-show-whitespace/archive/master.zip`) in `.zip` archive directly from GitHub and drag it onto the "Drag .zip here" area in the Extension Manager. Alternatively, use the "Install from URL..." link also in the Extension Manager.
 
-Instead of adding a separate `<div>` with an entry for the indentation of each line, the HTML code that CodeMirror itself produces is augmented. This makes the extension much more reliable and faster. Support for inline editors then is gratis, too - except for refresh calls when the setting is toggled.
+## Known Issues
 
+* CodeMirror overlays have a side effect of slowing down the editor in many aspects, including typing and scrolling. If the slowdown is too unbearable, it may be worthwhile to disable the whitespace except when required.
```
Comment (MiguelCastillo) [inline README.md]: "it may be worthwhile to disable the *extension* except when required."

File: package.json (hunk 5 above)
Comment (MiguelCastillo) [inline package.json]: Meh... I would sublime from this if possible.

**Comment by MiguelCastillo** [comment]
The changes look good!  A couple of tweaks.  But this is good to be merged. :)

**Comment by MiguelCastillo** [comment]
One thing I would consider doing is moving the license section to a file LICENSE file.  Just to reduce clutter from the readme.  And then push the changelog to the bottom

File: package.json (hunk 5 above)
Comment (le717) [inline package.json]: Ambiguous terms are ambiguous. :P You're saying to remove the notable changes highlight?

File: README.md (hunk 4 above)
Comment (le717) [inline README.md]: Fixed.

File: README.md (hunk 7 above)
Comment (le717) [inline README.md]: Fixed.

File: README.md (hunk 6 above)
Comment (le717) [inline README.md]: That feels a tad lazy to me, plus it kinda makes good sense to say upright how to install it (for those who do not know how).

**Comment by le717** [comment]
All changes pushed.

> One thing I would consider doing is moving the license section to a file LICENSE file. Just to reduce clutter from the readme.

Done.

> And then push the changelog to the bottom

I actually split the changelog off into a separate file in anticipation of adobe/brackets#10554 getting merged. :)

File: README.md (hunk 6 above)
Comment (MiguelCastillo) [inline README.md]: I can see that.  I guess I like consistency... If every extension pointed to the link then you wouldn't see little variations of the same stuff... :)  oh well.

**Comment by le717** [comment]
If all is well, time to merge!

**Pull request merged**